packages/core/content-manager/server/src/services/uid.ts

```diff
@@ -55,7 +55,7 @@ export default ({ strapi }: { strapi: Strapi }) => ({
   }) {
     const foundDocuments = await strapi.documents(contentTypeUID).findMany({
       filters: {
-        [field]: value,
+        [field]: { $contains: value },
```

Comment:
I don't quite understand the change, why does it need to use a contains?

Reply:
Previously we were doing an exact match on the proposed UID field value. We then iterate over the results adding '-${i}' as a suffix each time until we find one that is free.

So I think we really needed a fuzzy match on the proposed value so we collect all the potential clashes rather than just one

Reply:
I am trying to think of examples, but what if the uid field is:

`a`

And there is another one that is

`someting-with-a` 

wouldnt those collide ?

Reply:
ah I see! even if that matches, the proposed new uid should work correctly.
Do you think using "startWith" would be more appropiate? just to remove undesired matches